app/controllers/summoners_controller.rb

```diff
@@ -147,14 +184,76 @@ def stats_summary
             stats[:normal_team3][:stats] = game[i]
           end
         end
-        render json: {success: true, stats: stats}, status: 200
+        summoner.update(stats_summary: stats)
       end
     rescue OpenURI::HTTPError => e
       case rescue_me(e)
       when 1
         retry
       when 2
-        render json: {success: false}, status: 400
+        return e
+      end
+    end
+  end
+
+  def get_stats_ranked
+    summoner = Summoner.find_by(summonerId: params['id'].to_i, region: params['region'].to_s)
+    if summoner[:stats_ranked].nil?
+      update_stats_ranked(params['id'], params['region'], summoner)
+    end
+    # stats_ranked = summoner.stats_rankeds
+    # render json: {stats_ranked: summoner[:stats_ranked]}
+    render json: {success: true}
+  end
+
+  def renew_stats_ranked
+    summoner = Summoner.find_by(summonerId: params['id'], region: params['region'])
+    update_stats_ranked(params['id'], params['region'], summoner)
+    render json: {stats_ranked: summoner[:stats_ranked]}
+  end
```

Comment:
i want the user to manually update stuff unless there is no records of the user. The reason is that there is a request limit.
there is two ways to do it,
1) is the current way i have. Meaning all i need to do is to have $http use two different routes
2) is to merge these methods and have $http send in a parameter.
i think 1) is easier and faster to implement